rollouts/controllers/rollout_controller.go

```diff
@@ -510,11 +510,18 @@ func (r *RolloutReconciler) rolloutTargets(ctx context.Context, rollout *gitopsv
 
 	for _, target := range targets.ToBeCreated {
 		rootSyncSpec := toRootSyncSpec(target.packageRef)
+		var metadata *gitopsv1alpha1.Metadata
+
+		if rollout != nil && rollout.Spec.SyncTemplate != nil &&
+			rollout.Spec.SyncTemplate.RootSync != nil {
+			metadata = rollout.Spec.SyncTemplate.RootSync.Metadata
+		}
```

Comment:
If I understand it correctly, if one updates the `metadata` part of the syncTemplate, metadata will only be applied when creating new RootSync.

We should support the `Update` scenario as well. Somewhere there is a check where we decide target to be created or updated and it is on the basis of package `revision` field, we can extend it to compare the metadata part as well.

Reply:
Done.